cherab/core/atomic/zeeman.pyx

```diff
@@ -0,0 +1,124 @@
+DEF MULTIPLET_RATIO = 1
+
+DEF PI_POLARISATION = 0
+DEF SIGMA_POLARISATION = 1
+
+cdef class ZeemanStructure():
+    r"""
+    Provides wavelengths and ratios of
+    :math:`\pi`-/:math:`\sigma`-polarised Zeeman components for any given value of
+    magnetic field strength.
+
+    :param list wavelengths_pi: A list of Function1D objects that provide the wavelengths
+                                of individual :math:`\pi`-polarised Zeeman components for a given
+                                magnetic field strength.
+    :param list ratios_pi: A list of Function1D objects that provide the ratios
+                           of individual :math:`\pi`-polarised Zeeman components for a given
+                           magnetic field strength. The sum of all ratios must be equal to 1.
+    :param list wavelengths_sigma: A list of Function1D objects that provide the wavelengths
+                                   of individual :math:`\sigma`-polarised Zeeman components fo
+                                   a given magnetic field strength.
+    :param list ratios_sigma: A list of Function1D objects that provide the ratios
+                              of individual :math:`\sigma`-polarised Zeeman components for a given
+                              magnetic field strength. The sum of all ratios must be equal to 1.
+    """
+
+    def __init__(self, wavelengths_pi, ratios_pi, wavelengths_sigma, ratios_sigma):
+
+        if len(wavelengths_pi) != len(ratios_pi):
+            raise ValueError('The lengths of "wavelengths_pi" ({}) and "ratios_pi" ({}) do not match.'.format(len(wavelengths_pi),
+                                                                                                              len(ratios_pi)))
+
+        if len(wavelengths_sigma) != len(ratios_sigma):
+            raise ValueError('The lengths of "wavelengths_sigma" ({}) and "ratios_sigma" ({}) do not match.'.format(len(wavelengths_sigma),
+                                                                                                                    len(ratios_sigma)))
+
+        self._number_of_pi_lines = len(wavelengths_pi)
+        self._number_of_sigma_lines = len(wavelengths_sigma)
+
+        self._wavelengths = wavelengths_pi + wavelengths_sigma
+        self._ratios = ratios_pi + ratios_sigma
+
+        for i in range(len(self._wavelengths)):
+            self._wavelengths[i] = autowrap_function1d(self._wavelengths[i])
+            self._ratios[i] = autowrap_function1d(self._ratios[i])
+
+    @cython.boundscheck(False)
+    @cython.wraparound(False)
+    @cython.initializedcheck(False)
+    @cython.cdivision(True)
+    cdef double[:, :] evaluate(self, double b, bint polarisation):
+
+        cdef int i, start, number_of_lines
+        cdef np.npy_intp multiplet_shape[2]
+        cdef double ratio_sum
+        cdef np.ndarray multiplet
+        cdef double[:, :] multiplet_mv
+        cdef Function1D wavelength, ratio
+
+        b = max(0, b)
```

Comment:
Why is this done?

Reply:
Fixed to:
```python
if b < 0:  # safety check in case if used stand-alone
    raise ValueError('Argument "b" (magnetic field strength) must be non-negative.')
```